hud.lua

```diff
@@ -21,9 +21,13 @@ minetest.register_globalstep(function(dtime)
 		local areaStrings = {}
 
 		for id, area in pairs(areas:getAreasAtPos(pos)) do
-			table.insert(areaStrings, ("%s [%u] (%s%s)")
+			if area.faction_open and (areas.faction_available == false or factions.get_player_faction(area.owner) == nil) then
+				area.faction_open = false
```

Comment:
There could be one less faction getter call:
```Lua
local faction_info = area.faction_open and areas.faction_available and
	factions.get_player_faction(area.owner)
area.faction_open = type(faction_info) == "string"
...
		faction_info and (":" .. faction_info) or ""
```

Reply:
Good catch, this has been implemented